src/LiveQuery/QueryTools.js

```diff
@@ -206,7 +206,9 @@ function matchesKeyConstraints(object, key, constraints) {
         }
         break;
       case '$exists':
-        if (typeof object[key] === 'undefined') {
+        let propertyExists = typeof object[key] !== 'undefined';
+        let existenceIsRequired = constraints['$exists'];
```

Comment:
From my tests and understanding it will always be a boolean, because this query representation is done by the SDKs. It could be something different when the query is done via REST but this is not the case for LiveQuery right? 

Anyway the logic would still work if `constraints['$exists']` is `null/undefined`.